axelrod/fingerprint.py

```diff
@@ -0,0 +1,247 @@
+import axelrod as axl
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from itertools import product
+from axelrod.strategy_transformers import JossAnnTransformer, DualTransformer
+from axelrod.interaction_utils import compute_final_score_per_turn as cfspt
+from collections import OrderedDict
+
+
+def create_jossann(coordinate, probe):
+    """Creates a JossAnn probe player that matches the coordinate.
+
+    If the coordinate sums to more than 1 the parameters are flipped and
+    subtracted from 1 to give meaningful probabilities. This is outlined further
+    in [Ashlock2010]_.
+
+    Parameters
+    ----------
+    coordinate : tuple of length 2
+        coordinate of the form (x, y)
+    probe : class
+        A class that must be descended from axelrod.strategies
+
+    Returns
+    ----------
+    joss_ann: Joss-AnnTitForTat object
+        `JossAnnTransformer` with parameters that correspond to (x, y).
+    """
+    x, y = coordinate
+    if x + y > 1:
+        joss_ann = JossAnnTransformer((1 - y, 1 - x))(probe)()
+    else:
+        joss_ann = JossAnnTransformer((x, y))(probe)()
+    return joss_ann
+
+
+def create_coordinates(step):
+    """Creates a set of coordinates over the unit square.
+
+    Constructs (x, y) coordinates that are separated by a step equal to
+    `step`. The coordinates are over the unit squeare which implies
+    that the number of points created will be 1/`step`^2.
+
+    Parameters
+    ----------
+    step : float
+        The separation between each coordinate. Smaller steps will
+        produce more coordinates that will be closer together.
+
+    Returns
+    ----------
+    coordinates : list of tuples
+        Tuples of length 2 representing each coordinate, eg. (x, y)
+    """
+    coordinates = list(product(np.arange(0, 1, step),
+                               np.arange(0, 1, step)))
+    return coordinates
+
+
+def create_probes(probe, probe_coords):
+    """Creates a set of probe strategies over the unit square.
+
+    Constructs probe strategies that correspond to (x, y) coordinates. The
+    probes are created using the `JossAnnTransformer`.
+
+    Parameters
+    ----------
+    probe : class
+        A class that must be descended from axelrod.strategies.
+    probe_coords : list of tuples
+        Tuples of length 2 representing each coordinate, eg. (x, y)
+
+    Returns
+    ----------
+    probe_dict : ordered dictionary
+        An Ordered Dictionary where the keys are tuples representing each
+        coordinate, eg. (x, y). The value is a `JossAnnTransformer` with
+        parameters that correspond to (x, y).
+    """
+    probe_dict = OrderedDict((coord, create_jossann(coord, probe))
```

Comment:
Please feel free to comment as I'm going! Stops me from heading down the wrong path :)
I was just commenting to log thought process etc